.github/workflows/pr-check

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+import re
+import os
+import sys
+import json
+import argparse
+from typing import List, Optional
+
+import requests
+from github import Github
+from bs4 import BeautifulSoup
```

Comment:
Do we really need to parse html with beautiful soup? Surely the body is just markdown and we can find sections via a regex like `re.search('^#+ ', ..., flags=re.M)`?

Reply:
This is probably the most important design choice here;  glad to have this thread.

There's two decisions here:
1. should we parse markdown into a document tree or just wrangle raw text?
2. assuming parsing markdown: in what format do we want to work with the document tree?

**Re 1**: we kind of need to parse markdown properly (this is what commonmark is doing) because markdown can produce complex tree structures a la DOM. For example:

~~~markdown
## Change summary

Lorem ipsum, for example:

```python
from pydantic import BaseModel

## this is a Python comment in the code block, not a markdown section heading!
class MyModel(BaseModel):
    ...
```

In the meantime, the PR template has also changed and has a new comment in it

<!--
The comment is perfectly fine until it says something like:
* [ ] this, which should not be counted as an incomplete task
-->
~~~

We _could_ wrangle raw text and it will probably get lucky and seem fine most of the time until it breaks in confusing and unpleasant ways.

----

**Re 2**: This is a secondary choice. I only chose HTML (and beautifulsoup to work with it) for ease of use. The alternative would've been some other doc format or a markdown AST. The good thing about HTML is everyone knows about it and can reason about selectors, instead of some custom tree traversal logic.